Client/Frontend/Settings/HomePageSettingsViewController.swift

```diff
@@ -98,6 +98,16 @@ class WebPageSetting: StringSetting {
         textField.autocorrectionType = .No
     }
 
+    override func prepareValidValue(userInput value: String?) -> String? {
+        guard let value = value else {
+            return nil
+        }
+        if value.contains("://") {
+            return value
+        }
+        return "http://\(value)"
```

Comment:
How about `return URIFixup.getURL(value)?.absoluteString`?

In addition to adding a missing `http://`, `URIFixup` makes sure the URL is actually valid (by converting to an `NSURL`, triming extra whitespace, etc.)